bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/SensorBaseThingHandler.java

```diff
@@ -0,0 +1,282 @@
+        updateProperties(editProperties);
+
+        // Some sensors support optional channels
+        // (see https://github.com/dresden-elektronik/deconz-rest-plugin/wiki/Supported-Devices#sensors)
+        // any battery-powered sensor
+        if (sensorConfig.battery != null) {
+            createChannel(CHANNEL_BATTERY_LEVEL, ChannelKind.STATE);
+            createChannel(CHANNEL_BATTERY_LOW, ChannelKind.STATE);
+        }
+
+        createTypeSpecificChannels(sensorConfig, sensorState);
+
+        ignoreConfigurationUpdate = false;
+
+        // Initial data
+        updateChannels(sensorConfig);
+        updateChannels(sensorState, true);
+
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    protected void createChannel(String channelId, ChannelKind kind) {
+        ThingHandlerCallback callback = getCallback();
+        if (callback != null) {
+            ChannelUID channelUID = new ChannelUID(thing.getUID(), channelId);
+            ChannelTypeUID channelTypeUID;
+            switch (channelId) {
+                case CHANNEL_BATTERY_LEVEL:
+                    channelTypeUID = new ChannelTypeUID("system:battery-level");
+                    break;
+                case CHANNEL_BATTERY_LOW:
+                    channelTypeUID = new ChannelTypeUID("system:low-battery");
+                    break;
+                default:
+                    channelTypeUID = new ChannelTypeUID(BINDING_ID, channelId);
+                    break;
+            }
+            Channel channel = callback.createChannelBuilder(channelUID, channelTypeUID).withKind(kind).build();
+            updateThing(editThing().withoutChannel(channelUID).withChannel(channel).build());
+        }
+    }
+
+    public void valueUpdated(ChannelUID channelUID, SensorConfig newConfig) {
+        Integer batteryLevel = newConfig.battery;
+        switch (channelUID.getId()) {
+            case CHANNEL_BATTERY_LEVEL:
+                if (batteryLevel != null) {
+                    updateState(channelUID, new DecimalType(batteryLevel.longValue()));
+                }
+                break;
+            case CHANNEL_BATTERY_LOW:
+                if (batteryLevel != null) {
+                    updateState(channelUID, OnOffType.from(batteryLevel <= 10));
+                }
+                break;
+        }
+    }
+
+    public void valueUpdated(String channelID, SensorState newState, boolean initializing) {
+        switch (channelID) {
+            case CHANNEL_LAST_UPDATED:
+                String lastUpdated = newState.lastupdated;
+                if (lastUpdated != null && !"none".equals(lastUpdated)) {
+                    updateState(channelID,
+                            new DateTimeType(ZonedDateTime.ofInstant(
+                                    LocalDateTime.parse(lastUpdated, DateTimeFormatter.ISO_LOCAL_DATE_TIME),
+                                    ZoneOffset.UTC, ZoneId.systemDefault())));
+                }
+                break;
+        }
```

Comment:
```suggestion
            default:
                // other cases covered by sub-class
        }
```

For code readability

Reply:
Added